pennylane/measurements/expval.py

```diff
@@ -71,6 +71,20 @@ def return_type(self):
     def numeric_type(self):
         return float
 
+    def shape(self, device=None):
```

Comment:
General question: any reason to not use `@functools.lru_cache()` as it was done beforee?